### DISCUSSION TITLE
WARNING  No locally installed `turbo` found. Using version: 2.2.3. #NPM

### DISCUSSION THREAD

**Discussion by ghost** [vercel/turborepo#9385] [Help] [answered]
### Summary

I’ve created a Turbo Monorepo using npm and then added Docker following the provided [example](https://github.com/vercel/turborepo/tree/main/examples/with-docker). However, when building the project, I receive the warning message:
`WARNING  No locally installed `turbo` found. Using version: 2.2.3.`

Here is the complete log and Dockerfile for reference:

```docker

FROM node:22-alpine AS base

FROM base AS builder
RUN apk update
RUN apk add --no-cache libc6-compat
WORKDIR /app
RUN npm install turbo --global
COPY . .
RUN turbo prune --scope="web" --docker

FROM base AS installer
RUN apk update  
RUN apk add --no-cache libc6-compat
WORKDIR /app
COPY --from=builder /app/out/json/ .
RUN npm install
COPY --from=builder /app/out/full/ .

# ARG TURBO_TEAM
# ENV TURBO_TEAM=$TURBO_TEAM
# ARG TURBO_TOKEN
# ENV TURBO_TOKEN=$TURBO_TOKEN

RUN npm turbo build --filter=web...

FROM base AS runner
WORKDIR /app
RUN addgroup --system --gid 1001 nodejs
RUN adduser --system --uid 1001 nextjs
USER nextjs
COPY --from=installer --chown=nextjs:nodejs /app/apps/web/.next/standalone ./
COPY --from=installer --chown=nextjs:nodejs /app/apps/web/.next/static ./apps/web/.next/static
COPY --from=installer --chown=nextjs:nodejs /app/apps/web/public ./apps/web/public

CMD node apps/web/dist/index.js

```

![image](https://github.com/user-attachments/assets/f8a5d529-824c-411e-b85c-8056bd30f267)

**Comment by anthonyshew** [comment, marked as answer]
The warning is to inform you when the version of `turbo` meant for your repository hasn't been installed yet by your package manager. This is important to know in cases where a mismatch in global vs. local versions could cause errors. For instance, if you install a different major version.

 In this case, `turbo prune` is being ran from the global scope (`--global`), so it's not coming from the `node_modules` of your repository, so the warning is trying to tell you ahead of time of the possibility of a problematic mismatch in the future.
 
If you'd prefer to suppress the warning, you can use [the `TURBO_GLOBAL_WARNING_DISABLED` System Environment Variable](https://turbo.build/repo/docs/reference/system-environment-variables) can suppress the warning if you want to accept the risks and silence the warning. That would mean updating the code to `RUN TURBO_GLOBAL_WARNING_DISABLED=true turbo prune --scope="web" --docker`.

Note: I just went and looked around package manager documentation and it doesn't appear npm or pnpm have a way to install _one specific dependency_, which really is what you would need here.
